Mathlib/Topology/Separation.lean

```diff
@@ -129,6 +147,54 @@ theorem separatedNhds_iff_disjoint {s t : Set X} : SeparatedNhds s t ↔ Disjoin
 
 alias ⟨SeparatedNhds.disjoint_nhdsSet, _⟩ := separatedNhds_iff_disjoint
 
+/-- `SeparatingCover`s can be useful witnesses for `SeparatedNhds`. -/
+def SeparatingCover : Set X → Set X → Prop := fun s t ↦
+  ∃ u : ℕ → Set X, s ⊆ ⋃ n, u n ∧ ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) t
+
+/-- Used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+theorem separating_covers_iff_separated_nhds {h k : Set X} :
+    SeparatingCover h k ∧ SeparatingCover k h ↔ SeparatedNhds h k := by
+  constructor
+  · rintro ⟨⟨u, u_cov, u_props⟩, ⟨v, v_cov, v_props⟩⟩
+    have open_lemma : ∀ (u₀ a : ℕ → Set X), (∀ n, IsOpen (u₀ n)) →
+      IsOpen (⋃ n, u₀ n \ closure (a n)) := fun _ _ u₀i_open ↦
+      isOpen_iUnion (fun i ↦ IsOpen.sdiff (u₀i_open i) (isClosed_closure))
+    have cover_lemma : ∀ (h₀ : Set X) (u₀ v₀ : ℕ → Set X),
+        (h₀ ⊆ ⋃ n, u₀ n) → (∀ n, Disjoint (closure (v₀ n)) h₀) →
+        (h₀ ⊆ ⋃ n, u₀ n \ closure (⋃ m ∈ {m | m ≤ n}, v₀ m)) :=
+        fun h₀ u₀ v₀ h₀_cov dis x xinh ↦ by
+      rcases h₀_cov xinh with ⟨un , ⟨n, rfl⟩ , xinun⟩
+      simp only [mem_iUnion]
+      refine ⟨n, xinun, ?_⟩
+      rw [Set.Finite.closure_biUnion (finite_le_nat n)]
+      simp_all only [disjoint_right, mem_setOf_eq, mem_iUnion, exists_false, exists_const,
+        not_false_eq_true]
+    refine ⟨
+      ⋃ n : ℕ, u n \ (closure (⋃ m ∈ {m | m ≤ n}, v m)),
+      ⋃ n : ℕ, v n \ (closure (⋃ m ∈ {m | m ≤ n}, u m)),
+      open_lemma u (fun n ↦ ⋃ m ∈ {m | m ≤ n}, v m) (fun n ↦ (u_props n).1),
+      open_lemma v (fun n ↦ ⋃ m ∈ {m | m ≤ n}, u m) (fun n ↦ (v_props n).1),
+      cover_lemma h u v u_cov (fun n ↦ (v_props n).2),
+      cover_lemma k v u v_cov (fun n ↦ (u_props n).2),
+    ?_⟩
+    rw [Set.disjoint_left]
+    rintro x ⟨un, ⟨n, rfl⟩, xinun⟩
+    simp only [mem_iUnion, mem_diff, not_exists, not_and, Decidable.not_not]
+    intro m xinvgm
+    have n_le_m : n ≤ m := by
+      by_contra m_gt_n
+      exact xinun.2 (subset_closure (mem_biUnion (le_of_lt (not_le.mp m_gt_n)) xinvgm))
+    exact subset_closure (mem_biUnion n_le_m xinun.1)
+  · rintro ⟨U, V, U_open, V_open, h_sub_U, k_sub_V, UV_dis⟩
+    exact ⟨
+      ⟨fun _ ↦ U, Subset.trans h_sub_U (Eq.subset (Eq.symm (iUnion_const U))),
+      fun _ ↦ ⟨U_open, disjoint_of_subset (fun ⦃a⦄ a ↦ a) k_sub_V
+        (Disjoint.closure_left UV_dis V_open)⟩⟩,
+      ⟨fun _ ↦ V, Subset.trans k_sub_V (Eq.subset (Eq.symm (iUnion_const V))),
+      fun _ ↦ ⟨V_open, disjoint_of_subset (fun ⦃a⦄ a ↦ a) h_sub_U
+        (Disjoint.symm (Disjoint.closure_right UV_dis U_open))⟩⟩⟩
```

Comment:
The indentation of this block is pretty confusing. Also please use dot notation.

Reply:
Thoughts on 3419b6b ?